### PR TITLE
pre hook to make sure functional test dependencies are setup

### PR DESCRIPTION
**Pull request by mozrokafor**
## Description

    Test Chore: pre hook to setup test env

### CODE REVIEW

File: scripts/tests/prepare_tests.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+if [[ ! -f ".env" ]]
+then
+    echo ".env file does not exist"
+    touch .env
+    cat << 'EOF' > .env
+        export PATH=$PATH:~/Qt/6.2.4/macos/bin:$PATH
```
Comment (bakulf) [inline scripts/tests/prepare_tests.sh:7]: this works on your computer only :) Not on mine where:
- I use a different version of QT
- it's not installed in that path
- and it's not mac...

File: scripts/tests/prepare_tests.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        export QT_MACOS_BIN=~/Qt/6.2.4/macos/bin
+        MVPN_API_BASE_URL=http://localhost:5000
+        MVPN_BIN=build/tests/dummyvpn/dummyvpn
+        ARTIFACT_DIR=tests/artifact
+EOF
+fi
+
+# check if build directory exists
+if [[ ! -d "./build/tests/dummyvpn" ]]
```
Comment (bakulf) [inline scripts/tests/prepare_tests.sh:16]: note that 'build' is not a fix folder. Locally I don't use `build`.

File: scripts/tests/prepare_tests.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+then
+    echo "dummyvpn dir does not exist, run cmake command to build it"
+    cmake -B build .
```
Comment (bakulf) [inline scripts/tests/prepare_tests.sh:19]: I think the mocha beforeAll() function should check if the client exists and complain with a proper error message. Nothing more than this.

File: scripts/tests/prepare_tests.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+# install node modules to run functional tests if it doesn't exist
+if [[ ! -d "./node_modules" ]]
+then
+    echo "node modules does not exist, run npm install"
+    npm install
+fi
+
+# generate addons to run tests if they dont exist
+if [[ ! -d "./tests/functional/addons/01_empty_manifest" ]]
```
Comment (bakulf) [inline scripts/tests/prepare_tests.sh:42]: this is a test specific file. It should not be given as granted

**Review by bakulf — CHANGES_REQUESTED** [review]
I think all of this should (if we want - and I'm not 100% about it) in a `beforeAll()`

File: scripts/tests/prepare_tests.sh (hunk 1 above)
Comment (mozrokafor) [reply scripts/tests/prepare_tests.sh:7]: good catch, will remove

File: scripts/tests/prepare_tests.sh (hunk 3 above)
Comment (mozrokafor) [reply scripts/tests/prepare_tests.sh:19]: will remove the build command and put that in a string text

File: scripts/tests/prepare_tests.sh (hunk 4 above)
Comment (mozrokafor) [reply scripts/tests/prepare_tests.sh:42]: to my understanding this directory is a requirement to run the functional tests, if it does not exist should the dev trying to run tests be notified about the issue? better yet automatically generate the files for them?

**Comment by mozrokafor** [comment]
team, will be advised to setup an alias(or best alternative) on their machine to take care of this.

**Pull request closed without merging**